lib/orderbook/MatchingEngine.ts

```diff
@@ -220,8 +241,30 @@ class MatchingEngine {
     };
   }
 
-  public getPeerOrders = () => {
-    return this.getOrders(this.peerOrders);
+  public getPeerOrder = (orderId: string, peerPubKey: string): StampedPeerOrder => {
+    const peerOrders = this.peersOrders.get(peerPubKey);
+    if (!peerOrders) {
+      throw errors.PEER_ORDER_NOT_FOUND(orderId, peerPubKey);
+    }
+    const order = this.getOrder(orderId, peerOrders);
+    if (!order) {
+      throw errors.PEER_ORDER_NOT_FOUND(orderId, peerPubKey);
+    }
+    return order;
+  }
+
+  private getOrder = <T extends StampedOrder>(orderId: string, maps: OrderSidesLists<T>): T | undefined => {
+    return maps.buy.get(orderId) || maps.sell.get(orderId);
+  }
+
+  public getPeerOrders = (): OrderSidesArrays<StampedPeerOrder> => {
+    const res: OrderSidesArrays<StampedPeerOrder> = { buy: [], sell: [] };
+    this.peersOrders.forEach((peerOrders) => {
+      const peerOrdersArrs = this.getOrders(peerOrders);
+      res.buy.push(...peerOrdersArrs.buy);
+      res.sell.push(...peerOrdersArrs.sell);
```

Comment:
I believe we get a performance edge here using `concat` over `push` with the rest operator. I did a quick test and it seems to confirm that: http://jsben.ch/SZYnZ

You should be able to do `res.buy = res.buy.concat(peerOrdersArrs.buy);`.